### DISCUSSION TITLE
在安装VapourSynth-Preview时出现问题

### DISCUSSION THREAD

**Discussion by int-yscwl** [vcb-s/guides#14] [Q&A] [answered]
我用的是便携版 VS-C的portable 包，在看第五章 vapoursynth基础，（2）VapourSynth-Preview那里，运行 python -m vsstubs update时，会报Warning: Core only supports API R4.0 but the loaded plugin requires API R0.0这个错误，此外，用VSCode 打开 vpy 脚本使用 Ctrl+F6/Ctrl+F5时也会报这个错误，预览也无法正常开启，请问是哪里出了问题？？？

**Comment by sinsanction** [comment]
这个warning不影响功能，不能预览肯定是其它问题，有别的信息吗，最好带截图

**Reply by int-yscwl** [reply]
在vscode里，按Ctrl+F6没反应，而按Ctrl+F5时会报这个问题
![Snipaste_2024-04-29_11-12-20](https://github.com/vcb-s/guides/assets/66260414/3b3d4b7f-bf4c-489a-8f85-d5b5b388b206)
这是我写的简单脚本：
import vapoursynth as vs
core = vs.core
import havsfunc as haf
import mvsfunc as mvf

a = R"G:\XYJ\01\A1_t00.mkv"
src8 = core.lsmas.LWLibavSource(a)
src8.set_output()
请帮忙看看怎么回事，麻烦了

**Reply by sinsanction** [reply]
如果你是按照我们给的配置，你截图里的应该是Ctrl+F6 脚本检查，如果脚本无错误它确实什么都不做。
打印信息是Ctrl+F7
预览是Ctrl+F5，你截下预览时的截图

**Reply by int-yscwl** [reply]
首先是按Ctrl+F6，
![f6](https://github.com/vcb-s/guides/assets/66260414/adacfa19-28cf-44d6-a447-2d090be2c1ff)
不过哪怕故意写错，用脚本检查也没有任何反应；
其次是按Ctrl+F5，
![f5](https://github.com/vcb-s/guides/assets/66260414/68e10bde-f142-4a92-959f-a3df5d1d071a)
就会报出如图所示的错
最后是按Ctrl+F7，如按Ctrl+F6一样，也没有任何反应。
令我感到奇怪的是，在运行python -m vspreview xxx.vpy这条命令时，也成功的弹出类似 vsedit 的预览窗口，在装vscode时，扩展商店里也安装 Python 和 Pylance 扩展，以及VSCode 用户配置目录 %APPDATA%\Code\User\下也将本章 .vscode 目录的文件复制了
![Snipaste_2024-04-29_11-37-13](https://github.com/vcb-s/guides/assets/66260414/738f2b64-ae6c-4f50-ba25-595819d9487c)
settings.json 中的 python 路径也改了，改的是便携包里的路径：
![se](https://github.com/vcb-s/guides/assets/66260414/560f59e0-d7bc-4918-9f05-3dd6febd35b4)
按照教程再往下走，运行 python -m vsstubs update时就会报这个错：
![c](https://github.com/vcb-s/guides/assets/66260414/02b2a5f5-183c-48fd-96be-0af91555185b)
我不明白是我哪里错了……

**Reply by dtlnor** [reply]
前面說過了，这个warning不影响功能。並不是“出錯”。在代碼的世界裡warning是用來無視的（確信）

你keyboard binding.json 沒有放進去嗎

**Reply by int-yscwl** [reply]
是这个文件？
![Uploading key.jpg…]()

**Reply by int-yscwl** [reply]
![key](https://github.com/vcb-s/guides/assets/66260414/117f1489-eced-4d25-81b2-2b8f4aefbd9c)
刚刚图片好像出了问题，是这个文件吗

**Reply by dtlnor** [reply, marked as answer]
你應該在本repo該章節的[附件](https://github.com/vcb-s/guides/tree/master/Basics/%5B05%5D%20VapourSynth%E5%9F%BA%E7%A1%80/.vscode)裡找到這個文件，然後覆蓋掉你原本的

**Reply by int-yscwl** [reply]
我想我明白是哪里出了问题，安装VapourSynth-Plugins-Stub-Generator那里，我以为因为使用的是便携版，所以不需要把其.vscode目录下的文件复制到配置目录里，现在解决了，能正常检查和预览了，不过我并没有删除第一次settings.json里配置，而是复制进去了，应该不会有什么问题吧？

**Reply by sinsanction** [reply]
VSCode 和 VapourSynth 两块配置是完全独立的，便携版只是让你不用配置 VapourSynth 部分，我想教程里已经写的很明白哪一步可以不用配。
另外，你那半个 keybindings.json 又是哪来的（

**Reply by int-yscwl** [reply]
是在这个链接下载的：https://github.com/SaltyChiang/VapourSynth-Plugins-Stub-Generator/tree/R57Classic/.vscode

**Reply by int-yscwl** [reply]
另外，前面你提到的keyboard binding.json文件我找不到在哪

**Reply by dtlnor** [reply]
> 我想我明白是哪里出了问题，安装VapourSynth-Plugins-Stub-Generator那里，我以为因为使用的是便携版，所以不需要把其.vscode目录下的文件复制到配置目录里，现在解决了，能正常检查和预览了，不过我并没有删除第一次settings.json里配置，而是复制进去了，应该不会有什么问题吧？

便攜版自帶vsstubs，不需要你額外安裝。除非你用的不是便攜包。  
settings.json你可以一個個項自己研究一下起什麼作用，再決定如何把他融合到你自己原本的settings.json裡面。如果不確定的話，遇到同一個key，以本教程的附件裡面那個為準。

> 是在这个链接下载的：https://github.com/SaltyChiang/VapourSynth-Plugins-Stub-Generator/tree/R57Classic/.vscode

教程附件的.vscode跟vsstubs這個repo裡面的.vscode並不是一樣的。教程給的是教程的。你自己願意折騰配置的話可以參考repo裡面的。

> 另外，前面你提到的keyboard binding.json文件我找不到在哪

keyboard binding.json是我憑記憶寫出來的，寫錯了。

**Reply by int-yscwl** [reply]
我明白了，谢谢你的帮忙
